### PR TITLE
Indirect - ConvFit - Enable Plot Guess for Diffusion Functions

### LINKED ISSUE #13345: Allow Plot Guess for Diffusion Functions in Convfit
Currently you can not use the plot guess feature for any of the diffusion functions within the ConvFit tab. The check box for this is currently disabled in the interface as it is uncertain what the current state of plot guess for these function is.

### PR DESCRIPTION
**Pull request by brandonhewer**
Currently, in the ConvFit interface, the plot guess feature is disable for the Diffusion functions.

This PR enables plot guess and ensure it works as intended for all functions in ConvFit.

**To test:**
1. Navigate to Interfaces > Indirect > Data Analysis > ConvFit.
2. Supply the provided \*\_red file into the Sample field.
3. Supply the provided \*\_res file into the Resolution field.
4. For each fit function in the Fit Type drop-down menu:
    1. Select the fit type in the Fit Type drop-down menu.
    2.
        a) OPTIONAL (to get reasonable fit parameters) Click 'Run'
        b) Use plot guess to create a guess plot with the fit parameters.
    3. Ensure the guess is plotted (green plot) in the top preview plot (if 'Run' was clicked, guess plot should look very similar to fit plot).
    4. Move the blue fit selected in the top preview plot and ensure the guess plot updates accordingly.

**Must be merged after PR #20933**

**Test Files:**
[convfit-testfiles.zip](https://github.com/mantidproject/mantid/files/1401655/convfit-testfiles.zip)

Fixes #13345. 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by josephframsay — CHANGES_REQUESTED** [review]
Possibly I'm being a bit optimistic about how good a guess should be, but it seems to quite far off the mark in a few places. 

For example **StretchedExpFT** and **InelasticDiffSphere** are very short, and **Zero Lorentzians**, **InelasticDiffRotDiscreteCircle**, **ElasticDiffSphere** and **ElasticDiffRotDiscreteCircle** appear to be completely flat. Given that it's just a guess, is this an allowable amount of error?

Also, to what extent should the plot update when the blue bars are moved? For example, for **Two Lorentzians**, I would have expected the peak to get sharper as the bars draw in, but this isn't the case. Again, is this OK given that (I'm guessing) proper fitting gets done later on?

File: qt/scientific_interfaces/Indirect/IndirectDataAnalysisTab.cpp [hunk 1]
```diff
@@ -22,7 +24,7 @@ namespace IDA {
 IndirectDataAnalysisTab::IndirectDataAnalysisTab(QWidget *parent)
     : IndirectTab(parent), m_dblEdFac(nullptr), m_blnEdFac(nullptr),
       m_parent(nullptr), m_inputWorkspace(), m_previewPlotWorkspace(),
-      m_selectedSpectrum(0) {
+      m_selectedSpectra(0) {
```
Comment (josephframsay) [inline qt/scientific_interfaces/Indirect/IndirectDataAnalysisTab.cpp]: Does this want to be `Spectra` or `Spectrum`? It seems to me as though `m_selectedSpectrum` refers to a single spectrum, so calling it `Spectra` is confusing. Or am I missing something?

File: qt/scientific_interfaces/Indirect/IndirectDataAnalysisTab.cpp [hunk 2]
```diff
@@ -116,19 +150,190 @@ void IndirectDataAnalysisTab::plotCurrentPreview() {
...
+ *
+ * @param outputWS          The output workspace group.
+ * @param index             The index of the workspace (in the group)
+ *                          to plot.
+ * @param topPreviewPlot    The top preview plot.
+ * @param bottomPreviewPlot The bottom preview plot.
+ */
+void IndirectDataAnalysisTab::updatePlot(
+    WorkspaceGroup_sptr outputWS, size_t index,
+    MantidQt::MantidWidgets::PreviewPlot *topPreviewPlot,
```
Comment (josephframsay) [inline qt/scientific_interfaces/Indirect/IndirectDataAnalysisTab.cpp]: Given that the two plots have a specific purpose (the top is for sample and fit spectra, the bottom is for diffraction (?) spectra), could these have more descriptive names?

**Review by josephframsay — CHANGES_REQUESTED** [review]

File: qt/scientific_interfaces/Indirect/IndirectDataAnalysisTab.cpp (hunk 1 above)
Comment (SpencerHowells) [reply qt/scientific_interfaces/Indirect/IndirectDataAnalysisTab.cpp]: Some knowledge of the science is needed for these Guesses. Also, the data used is not appropriate for some of them. The purpose of plotting the Guess is to find suitable values by trial & error!
Moving the bars for the width in should sharpen up the peaks, though I have had problems previously with 2 Lorentzians.

**Review by josephframsay — APPROVED** [review]

**Comment by ianbush** [comment]
Release notes?

**Comment by brandonhewer** [comment]
@ianbush Changes have now been added to the release notes

**Comment by ianbush** [comment]
@brandonhewer A merge conflict has appeared for this now.

**Comment by brandonhewer** [comment]
@ianbush Merge conflicts have now been resolved

**Pull request merged**